src/apple/macos/disk.rs

```diff
@@ -1,200 +1,120 @@
-    }
-}
+pub(crate) fn get_disk_type(disk: &libc::statfs) -> DiskType {
+    let characteristics_string = CFReleaser::new(unsafe {
+        cfs::CFStringCreateWithBytesNoCopy(
+            kCFAllocatorDefault,
+            ffi::kIOPropertyDeviceCharacteristicsKey.as_ptr(),
+            ffi::kIOPropertyDeviceCharacteristicsKey.len() as isize,
+            cfs::kCFStringEncodingUTF8,
+            false as _,
+            kCFAllocatorNull,
+        )
+    })
+    .unwrap();
 
-pub(crate) fn get_disks(session: ffi::DASessionRef) -> Vec<Disk> {
-    if session.is_null() {
-        return Vec::new();
-    }
-    unsafe {
-        let count = libc::getfsstat(ptr::null_mut(), 0, libc::MNT_NOWAIT);
-        if count < 1 {
-            return Vec::new();
-        }
-        let bufsize = count * mem::size_of::<libc::statfs>() as c_int;
-        let mut disks = Vec::with_capacity(count as _);
-        let count = libc::getfsstat(disks.as_mut_ptr(), bufsize, libc::MNT_NOWAIT);
-        if count < 1 {
-            return Vec::new();
-        }
-        disks.set_len(count as _);
-        disks
-            .into_iter()
-            .filter_map(|c_disk| {
-                let mount_point = to_path(&c_disk.f_mntonname)?;
-                let disk = CFReleaser::new(ffi::DADiskCreateFromBSDName(
-                    kCFAllocatorDefault as _,
-                    session,
-                    c_disk.f_mntfromname.as_ptr(),
-                ))?;
-                let dict = CFReleaser::new(ffi::DADiskCopyDescription(disk.inner()))?;
-                // Keeping this around in case one might want the list of the available
-                // keys in "dict".
-                // core_foundation_sys::base::CFShow(dict as _);
-                let name = match get_str_value(dict.inner(), b"DAMediaName\0").map(OsString::from) {
-                    Some(n) => n,
-                    None => return None,
-                };
-                let removable =
-                    get_bool_value(dict.inner(), b"DAMediaRemovable\0").unwrap_or(false);
-                let ejectable =
-                    get_bool_value(dict.inner(), b"DAMediaEjectable\0").unwrap_or(false);
-                // This is very hackish but still better than nothing...
-                let type_ = if let Some(model) = get_str_value(dict.inner(), b"DADeviceModel\0") {
-                    if model.contains("SSD") {
-                        DiskType::SSD
-                    } else {
-                        // We just assume by default that this is a HDD
-                        DiskType::HDD
-                    }
-                } else {
-                    DiskType::Unknown(-1)
-                };
+    // Removes `/dev/` from the value.
+    let bsd_name = unsafe {
+        CStr::from_ptr(disk.f_mntfromname.as_ptr())
+            .to_bytes()
+            .strip_prefix(b"/dev/")
+            .expect("device mount point in unknown format")
+    };
+
+    let matching =
```

Comment:
Can you add a comment about this saying that there is no need to wrap it into a `CFReleaser` because `IOServiceGetMatchingServices` will take ownership of it please?